apps/src/studio/samBatLogic.js

```diff
@@ -0,0 +1,137 @@
+var studioConstants = require('./constants');
+var Direction = studioConstants.Direction;
+var Position = studioConstants.Position;
+var codegen = require('../codegen');
+var api = require('./api');
+
+/**
+ * Interface for a set of custom game logic for playlab
+ * @param {Studio} studio Reference to global studio object
+ * @interface CustomGameLogic
+ */
+function CustomGameLogic(studio) {}
+
+/**
+ * Logic to be run once per playlab tick
+ *
+ * @function
+ * @name CustomGameLogic#onTick
+ */
+
+/**
+ * Custom logic for the Sam the Bat levels
+ * @constructor
+ * @implements CustomGameLogic
+ */
+var SamBatLogic = function (studio) {
+  this.studio_ = studio;
+  this.cached_ = {};
+  this.samIndex = 0;
+};
+
+SamBatLogic.prototype.onTick = function () {
+  this.sam = this.studio_.sprite[this.samIndex];
+  
+  // Move Sam with arrow keys
+  for (var key in this.studio_.keyState) {
+    if (this.studio_.keyState[key] === 'keydown') { 
+      switch (key) {
+        case '37':
```

Comment:
If you add to the top of your file
```
var KeyCodes = require('../constants').KeyCodes;
```
You can now do KeyCodes.LEFT, etc instead of referencing the key directly